host.go

```diff
@@ -46,6 +47,15 @@ type gpbftRunner struct {
 	runningCtx context.Context
 	errgrp     *errgroup.Group
 	ctxCancel  context.CancelFunc
+
+	// msgsMutex guards access to selfMessages
+	msgsMutex    sync.Mutex
+	selfMessages map[uint64]map[roundPhase]*gpbft.GMessage
```

Comment:
There can be multiple messages for a single (instance, round, phase) as we can have participating IDs.